kv/memberlist/tcp_transport.go

```diff
@@ -358,13 +360,11 @@ func (t *TCPTransport) FinalAdvertiseAddr(ip string, port int) (net.IP, int, err
 			return nil, 0, fmt.Errorf("failed to parse advertise address %q", ip)
 		}
 
-		// Ensure IPv4 conversion if necessary.
-		if ip4 := advertiseAddr.To4(); ip4 != nil {
```

Comment:
I think this is no longer required, since if the parse fails above, we will have already returned, so dropped it here.